qcfractal/storage_sockets/models.py

```diff
@@ -0,0 +1,259 @@
+import mongoengine as db
+import datetime
+
+
+class Collection(db.DynamicDocument):
+    """
+        A collection of precomuted workflows such as datasets, ..
+
+        This is a dynamic document, so it will accept any number of
+        extra fields (expandable and uncontrolled schema)
+    """
+
+    collection_type = db.StringField(default='', required=True,
+                                     choices=['dataset', '?'])
+    name = db.StringField(default='', required=True)  # Example 'water'
+
+    meta = {
+        'collection': 'collections',  # DB collection/table name
+        'indexes': [
+            {'fields': ('collection_type', 'name'), 'unique': True}
+        ]
+    }
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+class Molecule(db.DynamicDocument):
+    """
+        The molecule DB collection is managed by pymongo, so far
+    """
+
+    name = db.StringField()
+    molecular_formula = db.StringField()
+    molecule_hash = db.StringField()
+
+    def create_hash(self):
+        """ TODO: create a special hash before saving"""
+        return ''
```

Comment:
Is there any way to get the full JSON document here? Should simple to do so if so.

Also should we call this `Molecule` as there is a `Molecule` class in interface.

Reply:
Yes, it's a to_json() method. We always want to work with the class itself unless we will send it over the REST API.

For the molecule naming, I think there should be no issue in the future. Portal won't shouldn't import or use the storage package at all. Levi's classes should be in another package used by Portal and not by Fractal.